### PR TITLE
[FLINK-34250] add config docs for formats

### PR DESCRIPTION
**Pull request by HuangZhenQiu**
## What is the purpose of the change

Generate the doc sections for Flink format options.

## Brief change log
1. Add format packages ConfigurationOptionLocator
2. Generated html files for formats
3. config.md to link all of the generated config html files.
4. Change the class ConfigOptionsDocsCompletenessITCase to allow duplicated key ignore-parse-errors in both json and csv format.

## Verifying this change
1. Completeness is covered by ConfigOptionsDocsCompletenessITCase

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (docs)

### CODE REVIEW

**Comment by wuchong** [comment]
@leonardBang do you have time to help review this?

File: docs/content/docs/deployment/config.md [hunk 1]
```diff
@@ -479,6 +479,26 @@ Flink does not use Pekko for data transport.
 
 {{< generated/akka_configuration >}}
 
+### Flink Format Options
```
Comment (PatrickRen) [inline docs/content/docs/deployment/config.md:482]: I'm afraid this is not the right place to list format options. This page shows all configurations that can be used in `conf/flink-conf.yaml`(now renamed to `conf/config.yaml`), which controls the behavior of Flink cluster. Formats are only used in some Table / SQL connectors such as file system and Kafka, and there are already pages for them. Take CSV as an example:

https://github.com/apache/flink/blob/558ca75da2fcec875d1e04a8d75a24fd0ad42ccc/docs/content/docs/connectors/table/formats/csv.md

**Review by PatrickRen — COMMENTED** [review]
@HuangZhenQiu Thanks for the contribution! I left a comment

File: docs/content/docs/deployment/config.md (hunk 1 above)
Comment (HuangZhenQiu) [reply docs/content/docs/deployment/config.md:482]: Thanks for the info. If so, I think this PR is not really needed. I will close it.

**Pull request closed without merging**